packages/@lwc/engine-core/src/framework/base-lightning-element.ts

```diff
@@ -542,6 +544,18 @@ function warnIfInvokedDuringConstruction(vm: VM, methodOrPropName: string) {
         return vm.shadowRoot;
     },
 
+    get elementSelf(): Element {
+        const vm = getAssociatedVM(this);
+
+        if (process.env.NODE_ENV !== 'production') {
+            if (isFalse(vm.elm instanceof Element)) {
+                logError(`this.elementSelf should be an Element, found: ${vm.elm}`);
```

Comment:
I think we should do `assert.false` here instead.

Historically we _do_ try to avoid throwing dev-only errors, but in this case I don't think it applies, because this is an invariant in the framework itself. If we are returning something other than an `Element` here, then something has gone horribly wrong and we need to fix it.

Reply:
sure, let me have a look on it